app/models/framework_response/details_object.rb

```diff
@@ -6,9 +6,9 @@ class DetailsObject
 
     attr_accessor :question_options, :details_options, :option, :details
 
-    validates :option, presence: true
-    validates :option, inclusion: { in: :question_options }, if: :question_options
-    validates :details, presence: true, if: :details_options
+    validates :option, presence: true, if: :details
+    validates :option, inclusion: { in: :question_options }, if: ->(object) { object.question_options && object.option.present? }
```

Comment:
What type is `object` here? I'm wondering if some of this logic expression could be moved to the relevant class and given a method name to make the intent more obvious.